vtex-sites/base.store
# chore: Improve overall performance

## What's the purpose of this pull request?
#295 added login functionality. This, however, introduced an unnecessary re-render of the whole app. This PR removes the need for this re-render, thus reducing TBT and increasing overall performance

## How does it work?
#295 made the return type of the `person` query to be `StorePerson!`. This means that the query result of person returns the following value for the anonymous user:
```json
{
  "id": "",
  "email": "",
  "givenName": "",
  "familyName": ""
}
```

This, in turn, is different from the default user session, passed in the [context provider](https://github.com/vtex-sites/base.store/blob/0a16a3784749af1f60bc359d7a5cbafb8b9783ef/gatsby-browser.js#L29) as `null`. This mismatch in values, caused the session to be reset for the anonymous user, causing an unnecessary re-render.

To fix this issue, I made the api return `null` for the `person` query when executed by an anonymous user. This matches the default value from the session, this not triggering a re-render. All changes are performed on this [PR](https://github.com/vtex/faststore/pull/1190) at faststore

An evidence can be seen below were the first image shows the `ProductGallery` component being rendered twice (due to session being reset) and the second image shows the absence of this re-render after the proposed fix. 

<img width="1680" alt="image" src="https://user-images.githubusercontent.com/1753396/159959551-1c32429d-9bee-4921-b8ad-543273c42888.png">
<img width="1680" alt="image" src="https://user-images.githubusercontent.com/1753396/159960376-86b61a4d-e932-4daa-b6d3-ccd33a183ab7.png">

## How to test it?
Make sure everything works as before and `ProductGallery` renders once. 
> Make sure to delete any pre-stored value in indexedDB

## Checklist
